### ISSUE TITLE
calculateProportionalAmounts for boosted pools

### ISSUE THREAD

**Issue by agualis** [balancer/b-sdk#468]
The FE we uses `calculateProportionalAmounts`  when the user choses “Proportional addable pool tokens”.
They fill one token input and we auto calculate and fill the other inputs proportionally using `calculateProportionalAmounts` that will be used with `kind: AddLiquidityKind.Unbalanced` underneath.

From a DX point of view, it would be easier to make it work with underlying tokens as the rest of the boosted integration.

**Comment by brunoguerios** [comment]
Hey @agualis !
I'm wondering if the SDK is the best place to adapt this logic 🤔 
The helper itself is simply calculating proportional amounts given a reference amount, while the FE is handling pool state by fetching/managing `poolStateWithBalances`.
If you use underlying token amounts as input, it should work as is.
Looking at the FE code, it seems to me that within `_calculateProportionalHumanAmountsIn` you could shift between `poolStateWithBalances` and something like `boostedPoolStateWithBalances` accordingly.
Does that make sense?

cc @johngrantuk (any idea/feedback on the suggested approach?)

**Comment by agualis** [comment]
@johngrantuk @brunoguerios I'm ok with the suggestion and with trying to improve the DX after V3 launch

**Comment by brunoguerios** [comment]
Yeah, my concern is that if we shift that to the SDK, it will have to perform a few on-chain calls to query the underlying balances and then calculate amounts proportional to it.
While current helper expects the pool state (balances) to be provided as input.
I'm actually implementing something like that, with on-chain calls to solve a similar issue.
I'll share with you after I'm done so we can evaluate if some of that is reusable into solving/improving this.

**Comment by brunoguerios** [comment]
@agualis - sharing in case it's helpful:

PR #486 adds the concept of `PoolStateWithUnderlyingBalance` and also shows how you can adapt it to `PoolStateWithBalance` to reuse it within `calculateProportionalAmounts`.

Let me know if you think any of this can be used/adapted to improve DX after v3 launch
